File: nim/100_doors.nim
import std/enumerate

proc get_doors(): array[100, bool] =
  var doors: array[100, bool]

  for n in 1..100:
    for i, door in enumerate(doors):
      if i mod n == 0:
        doors[i] = not doors[i]
  return doors

proc display(items: array[100, bool]) =
  for i, item in enumerate(items):
    if item: stdout.write i, " "

proc main() =
  var nums = get_doors()
  display(nums)

main()

File: c++/100_doors.cpp
#include <iostream>

int main() {
    bool doors[100] = {false};

    for (int i = 1; i < 101; i++) {
        for (int j = 0; j < 100; j++) {
            if (j % i == 0) {doors[j] = !doors[j];}
        }
    }

    for (int i = 0; i < 100; i++) {
        if (doors[i]) {std::cout << i << " ";}
    }
    return 0;
}
